src/blockrelay/graphene.cpp

```diff
@@ -1509,3 +1511,10 @@ uint64_t GetShortID(uint64_t shorttxidk0, uint64_t shorttxidk1, const uint256 &t
     static_assert(SHORTTXIDS_LENGTH == 8, "shorttxids calculation assumes 8-byte shorttxids");
     return SipHashUint256(shorttxidk0, shorttxidk1, txhash) & 0xffffffffffffffL;
 }
+
+// Generate cheap hash from seeds using SipHash
+uint64_t GetShortID(uint64_t shorttxidk0, uint64_t shorttxidk1, const uint256& txhash) {
+    //return txhash.GetCheapHash();
```

Comment:
since GetShortID isn't static and GetShortID is also defined in compactblocks.cpp/h.  I think it would be better to either make the distinction here and call this GetGrShortID() or perhaps better , combine the two ...create a common GetShortID() and put it in blockrelay_common and pass the byte size you want since compactblocks uses 6 and graphene uses 8.

Reply:
I'm not strongly opposed to putting `GetShortID` in blockrelay_common, but in addition to differing byte count, graphene also need to do a version check. So the only line that remains the same is the last one. For that reason, I lean toward only renaming it to `GetGrShortID` as you suggested.